specfile/tags.py

```diff
@@ -0,0 +1,445 @@
+    "Sha1header",
+    "Sha256header",
+    "Sig_base",
+    "Siggpg",
+    "Sigmd5",
+    "Sigpgp",
+    "Sigsize",
+    "Size",
+    "Source",
+    "Sourcepackage",
+    "Sourcepkgid",
+    "Sourcerpm",
+    "Suggestflags",
+    "Suggestname",
+    "Suggestnevrs",
+    "Suggests",
+    "Suggestversion",
+    "Summary",
+    "Supplementflags",
+    "Supplementname",
+    "Supplementnevrs",
+    "Supplements",
+    "Supplementversion",
+    "Transfiletriggerconds",
+    "Transfiletriggerflags",
+    "Transfiletriggerin",
+    "Transfiletriggerindex",
+    "Transfiletriggername",
+    "Transfiletriggerpostun",
+    "Transfiletriggerpriorities",
+    "Transfiletriggerscriptflags",
+    "Transfiletriggerscriptprog",
+    "Transfiletriggerscripts",
+    "Transfiletriggertype",
+    "Transfiletriggerun",
+    "Transfiletriggerversion",
+    "Triggerconds",
+    "Triggerflags",
+    "Triggerin",
+    "Triggerindex",
+    "Triggername",
+    "Triggerpostun",
+    "Triggerprein",
+    "Triggerscriptflags",
+    "Triggerscriptprog",
+    "Triggerscripts",
+    "Triggertype",
+    "Triggerun",
+    "Triggerversion",
+    "Url",
+    "V",
+    "Vcs",
+    "Vendor",
+    "Verbose",
+    "Verifyscript",
+    "Verifyscriptflags",
+    "Verifyscriptprog",
+    "Veritysignaturealgo",
+    "Veritysignatures",
+    "Version",
+    "Xpm",
+]
+
+
+class Tag:
+    """
+    Class that represents spec file tag.
+
+    Attributes:
+        name: Name of the tag.
+        value: Literal value of the tag as stored in the spec file.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        value: str,
+        expanded_value: str,
+        separator: str,
+        preceding_lines: List[str],
+    ) -> None:
+        if not name or name.capitalize().rstrip("0123456789") not in TAG_NAMES:
+            raise ValueError(f"Invalid tag name: '{name}'")
+        self.name = name
+        self.value = value
+        self._expanded_value = expanded_value
+        self._separator = separator
+        self._preceding_lines = preceding_lines.copy()
+
+    def __repr__(self) -> str:
+        preceding_lines = repr(self._preceding_lines)
+        return (
+            f"Tag({self.name}, '{self.value}', '{self._expanded_value}', "
+            f"'{self._separator}', {preceding_lines})"
+        )
+
+    @property
+    def valid(self) -> bool:
+        """Validity of the tag. A tag is valid if it 'survives' the expansion of the spec file."""
+        return self._expanded_value is not None
+
+    @property
+    def expanded_value(self) -> str:
+        """Value of the tag after expanding macros and evaluating all conditions."""
+        return self._expanded_value
+
+    @property
+    def comments(self) -> List[str]:
+        """List of comment lines associated with the tag."""
+        comments = itertools.takewhile(
+            lambda x: x.startswith("#"), reversed(self._preceding_lines)
+        )
+        return [c[1:] for c in reversed(list(comments))]
```

Comment:
Because the result is supposed to be a list of consecutive comment lines directly above the tag line, so it is necessary to iterate from the bottom up.